Feedback items (set 2)

shubham-dreamspring opened the pull request:
- Change name of car.controller to carController
- Make model ParkingLot
- Remove use of orm from controllers
- Make methods for find cars, add car, remove car in Car model
- Make methods for park and unpark in ParkingLot model
- Make wrapper method for orm methods

shubham-dreamspring committed:
Refactor orm.js
-
- Add a method to read and write operations
- Remove read and write logic from every other method in orm.js

shubham-dreamspring committed:
Add empty lines after every logical units

shubham-dreamspring committed:
Add success message in parkCar method
-
- Add success message in car.controller.js
- Render the success message from server in client side js

shubham-dreamspring committed:
Refactor Car Model
-
- Seperate the logic in ParkingLot Model and Car Model
- Change controller use appropriate model

shubham-dreamspring committed:
Rename files and sycn the code

- Rename car.controller to ParkingLotController
- Rename index.controller to indexController
- Change car.js to parkingLot.js
- Rename route /cars to / parkinglot

shubham-dreamspring committed:
Remove access of orm from controller
-
- Shift logic of find cars in Car model
- Remove redundant api for getting empty slots

shubham-dreamspring committed:
Sync the changes in tests

jinifrancis commented on controller/parkingLotController.js:
```diff
@@ -1,48 +1,45 @@
+class ParkingLotController {
   getAllCars(_, res, __) {
-    let data = orm.findAll("car");
+    let data = Car.findAllCars();
```
Rename to `findAll()`

jinifrancis replied:
Where is the method `findAllCars()` defined?

jinifrancis commented on model/car.js:
```diff
@@ -1,56 +1,64 @@
-    } else {
-      throw new Error("Invalid Registration no");
-    }
+    return validator.validateRegNo(this.registration_no);
```
Rename to `isValidRegistrationNumber`

jinifrancis commented on model/car.js:
```diff
@@ -1,56 +1,64 @@
   }
 
-  park() {
+  isRegistrationvalid() {
```
Method is not in the camelCase format completely

jinifrancis commented on model/car.js:
```diff
@@ -1,56 +1,64 @@
 
+  isParked() {
+    const orm = new CustomORM();
+    const car = orm.findById("car", "registration_no", this.registration_no);
```
Can it be refactored in such a way that it gets the details like `Car.findBy('registration_no', this.registration_no)`

jinifrancis commented on model/car.js:
```diff
@@ -1,56 +1,64 @@
+      this.slot = car.slot;
+      this.park_timestamp = car.park_timestamp;
+    }
     return car;
```
When the function name starts with `is` it is expected that the return value is a boolean by convention. 

Also, I notice that you are setting the slot and park_timestamp to the current car object, and you are returning another object.

jinifrancis commented on model/car.js:
```diff
@@ -1,56 +1,64 @@
     }
+    this.slot = slot;
+    return slot;
+  }
```
This function has multiple responsibilities:
1. Return the slot number of an existing car
2. Find an empty slot in the parking lot
3. Set a new slot number to a car

Can you refactor in such a way that each of these functionalities have its own method?
Ex:
`car.getSlot()`
`parkingLot.getEmptySlot()`
`car.setSlot()`

jinifrancis commented on model/car.js:
```diff
@@ -1,56 +1,64 @@
+      slot: this.slot,
+      park_timestamp: this.park_timestamp,
+    };
+    return orm.pushData("car", car);
```
Can we do `orm.pushData("car", this.allAttributes)` instead of creating a new object using existing details? The `allAttributes()` would be a new method

jinifrancis commented on model/car.js:
```diff
@@ -1,56 +1,64 @@
 class Car {
-  constructor(registration_no) {
+  constructor(registration_no, slot = null) {
     this.registration_no = registration_no;
```
camelCase variable names

jinifrancis commented on model/parkingLot.js:
The variable (object of class Car) passed as parameter to the methods in this class should be `car` and not `Car`

jinifrancis commented on model/parkingLot.js:
```diff
@@ -0,0 +1,46 @@
+      registration_no: Car.registration_no,
+      slot: Car.slot,
+      park_timestamp: Car.park_timestamp,
+    };
```
Can't we return the `car` object directly instead of creating a new object with the same attributes as that of `car`? Else can you think of another method `car.allAttributes()` like mentioned on a previous comment?

jinifrancis commented on routes/index.js:
```diff
@@ -1,8 +1,10 @@
 let indexController = new IndexController();
+
 /* GET home page. */
 router.get("/",indexController.getIndex);
```
Missing space after `,` in the parameter list

jinifrancis commented on spec/validator.spec.js:
```diff
@@ -1,12 +1,15 @@
 var Validator = require("../utils/validator");
 
 describe("Vadiation of registration number", () => {
+  
```
Remove new line

jinifrancis commented on app.js:
```diff
@@ -6,7 +6,7 @@ var logger = require("morgan");
-var carsRouter = require("./routes/cars");
+var parkingLotRouter = require("./routes/parkingLot");
 
 var app = express();
```
Why are we using `var` here?

jinifrancis commented on routes/parkingLot.js:
```diff
@@ -0,0 +1,19 @@
+const ParkingLotController = require("../controller/parkingLotcontroller");
+
+const parkingLotController = new ParkingLotController();
+var router = express.Router();
```
Why `var`?

jinifrancis commented on routes/parkingLot.js:
```diff
@@ -0,0 +1,19 @@
+const parkingLotController = new ParkingLotController();
+var router = express.Router();
+
+/* GET users listing. */
```
Is this comment still relevant?

jinifrancis commented on utils/validator.js:
```diff
@@ -1,4 +1,5 @@
 class Validator {
+  
```
Remove new line

shubham-dreamspring committed:
Refactor code

- Replace var with const
- Correct method call findAll cars in controller
- Camelcase registration_no
- Rename isRegistranValid
- Change logic of isParked
- Fetch car details in constructor
- Make getter and setter of slot in Car model
- Make a method to get empty slot in parking lot
- Change name of validRegNo in Validator
- Sync the changes in tests
